File: Practice/gcd_lcm.py
def gcd(a, b, smol):
    if a % smol == 0 and b % smol == 0:
        return smol
    else:
        return gcd(a, b, smol - 1)


def lcm(a, b, lorg):
    if lorg % a == 0 and lorg % b == 0:
        return lorg
    else:
        return lcm(a, b, lorg + 1)


x = int(input("Enter 1st number: "))
y = int(input("Enter 2nd number: "))
small = x if x < y else y
large = x if x > y else y
print(f"GCD = {gcd(x, y, small)}")
print(f"LCM = {lcm(x, y, large)}")

File: Lab Practice/gcd_lcm.c
#include <stdio.h>
int gcd(int x, int y, int small)
{
	for (int i = small; i > 0; i--)
	{
		if (x % i == 0 && y % i == 0)
		{
			small = i;
			break;
		}
	}
	return small;
}
int hcf(int x, int y, int small)
{
	if (x % small == 0 && y % small == 0)
	{
		return small;
	}
	else
	{
		return hcf(x, y, small - 1);
	}
}
int lcm(int x, int y, int large)
{
	for (int i = large; i <= x * y; i++)
	{
		if (i % x == 0 && i % y == 0)
		{
			large = i;
			break;
		}
	}
	return large;
}
int main()
{
	int num1, num2, smol, lorge;
	printf("1st number: ");
	scanf("%d", &num1);
	printf("2nd number: ");
	scanf("%d", &num2);

	smol = (num1 <= num2) ? num1 : num2;
	lorge = (num1 >= num2) ? num1 : num2;

	int a = hcf(num1, num2, smol);
	int b = lcm(num1, num2, lorge);

	printf("GCD of %d and %d is %d\n", num1, num2, a);
	printf("LCM of %d and %d is %d", num1, num2, b);
	return 0;
}